drivers/gpio/gpio_esp32.c

```diff
@@ -355,6 +428,9 @@ GPIO_DEVICE_INIT(DT_INST_1_ESPRESSIF_ESP32_GPIO_LABEL, 32_to_39);
 
 static void gpio_esp32_isr(void *param)
 {
+	/* FIXME: There is a problem in clearing interrupts
+	 *  of type LEVEL_LOW/HIGH, hence it is not usable
```

Comment:
We shouldn't go too deeply into the details of the issue, however the fact that the level interrupts cannot be cleared is the expected behavior. The unexpected behavior is that they are not re-triggered, despite that the interrupt status bit set high the interrupt routine is not invoked.

Reply:
> The unexpected behavior is that they are not re-triggered, despite that the interrupt status bit set high the interrupt routine is not invoked.

The problem is the new interrupt won't be triggered until the status flag is cleared, and  this clearing is not done properly!.

Here are what i tried to do:
- Write 0x0000 0000 to STATUS_REG directly
- Write 0xFFFF FFFF to W1TC_REG
- Write 0xFFFF FFFF to W1TC_REG using assembly instructions
- shorten the ISR to just clear the STATUS flag (no function calls at all).

Other thing i noticed in another implementations is, all of them include **a loop**  to check and clear interrupt status for each GPIO. this looks weird to me because ISRs should be in general small, fast and for sure no loops at all.

I don't know if @andyross faced something similar while adding support for xtensa architecture.